homeassistant/components/touchlinesl/climate.py

```diff
@@ -0,0 +1,119 @@
+"""Roth Touchline SL climate integration implementation for Home Assistant."""
+
+from typing import Any
+
+from pytouchlinesl import Zone
+
+from homeassistant.components.climate import (
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.const import ATTR_TEMPERATURE, UnitOfTemperature
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import TouchlineSLConfigEntry
+from .const import DOMAIN
+from .coordinator import TouchlineSLModuleCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: TouchlineSLConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Touchline devices."""
+    coordinators = entry.runtime_data
+    for module_id in coordinators:
+        async_add_entities(
+            TouchlineSLZone(coordinator=coordinators[module_id], zone_id=z)
+            for z in coordinators[module_id].data.zones
+        )
```

Comment:
```suggestion
    async_add_entities(
        TouchlineSLZone(coordinator=coordinators, zone_id=zone_id)
        for zone_id in coordinators.data.zones
        for coordinator in coordinators
    )
```
(I personally don't like the kwargs, since it's now like practically the same)